tensorboard/components/vz_line_chart2/linear-scale.ts

```diff
@@ -0,0 +1,119 @@
+/* Copyright 2018 The TensorFlow Authors. All Rights Reserved.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+==============================================================================*/
+namespace vz_line_chart2 {
+
+export class LinearScale extends Plottable.Scales.Linear implements ITfScale {
+  private _ignoreOutlier: boolean = false;
+  protected _valueProviderForDomain: ValueProviderForDomain;
+
+  constructor() {
+    super();
+    this.padProportion(.2);
+  }
+
+  public setValueProviderForDomain(provider: ValueProviderForDomain): this {
+    this._valueProviderForDomain = provider;
+    return this;
+  }
+
+  /**
+   * Adds some padding to a given domain. Specifically, it:
+   * - returns about [-0.1a, 2.1a] when a = b and a >= 0.
+   * - returns about [-2.1a, 0.1a] when a = b and a < 0.
+   * - returns [-0.1b, b + padProportion * (b-a)] if b > 2a and a > 0
+   * - else, pads by `padProportion`
+   * @override
+   */
+  protected _niceDomain(domain: number[], count?: number): number[] {
+    const [a, b] = domain;
+    let padding: number;
+    const span = b - a;
+    if (span === 0) {
+      // If b===a, we would create an empty range. We instead select the range
+      // [0, 2*a] if a > 0, or [-2*a, 0] if a < 0, plus a little bit of
+      // extra padding on the top and bottom of the plot.
+      padding = Math.abs(a) * 1.1 + 1.1;
```

Comment:
Not answering your question directly, I took the code directly from
https://github.com/tensorflow/tensorboard/blob/master/tensorboard/components/vz_chart_helpers/vz-chart-helpers.ts#L154-L178

Instead of focusing on correctness, I made sure the linear scale remains exactly as is today. I honestly do not exactly understand the original author's intension here.